packages/http-server-csharp/test/generation.test.ts

```diff
@@ -1098,3 +1104,89 @@ it("handles implicit request body models correctly", async () => {
     ],
   );
 });
+
+it("handles multipartBody requests and shared routes", async () => {
+  await compileAndValidateMultiple(
+    runner,
+    `
+      model Bar<T extends {}> {
+        ...T;
+      }
+      model FooRequest {
+        contents: HttpPart<File>;
+        other: HttpPart<Bar<FooJsonRequest>>;
+      }
+      model FooJsonRequest {
+        mediaType: string;
+        filename: string;
+        contents: bytes;
+      }
+
+      @sharedRoute
+      @route("/foo") 
+      @post 
+      op fooBinary(
+        @header("content-type") contentType: "multipart/form-data", 
+        @multipartBody body: FooRequest
+      ): void;
+
+      @sharedRoute
+      @route("/foo") 
+      @post 
+      op fooJson(
+        @header("content-type") contentType: "application/json", 
+        @body body: FooJsonRequest
+      ): void;
+      `,
+    [
+      [
+        "FooJsonRequest.cs",
+        [
+          "public partial class FooJsonRequest",
+          "public string MediaType { get; set; }",
+          "public string Filename { get; set; }",
+          "public byte[] Contents { get; set; }",
+        ],
+      ],
+      [
+        "ContosoOperationsControllerBase.cs",
+        [
+          "using Microsoft.AspNetCore.WebUtilities;",
+          "using Microsoft.AspNetCore.Http.Extensions;",
+          `[Consumes("multipart/form-data")]`,
+          "public virtual async Task<IActionResult> FooBinary(HttpRequest request, Stream body)",
+          ".FooBinaryAsync(reader)",
```

Comment:
nit, is this supposed to be complete compilable code or substring to pass into template that will form a valid c# file?

Reply:
These are validations of important features of the generated code
- it uses the consumes header to route traffic
- It has the right operation signature
- It calls the right method in the interface